frontend/src/scenes/data-management/events/eventDefinitionsTableLogic.ts

```diff
@@ -0,0 +1,137 @@
+import { kea } from 'kea'
+import { AnyPropertyFilter, EventDefinition, PropertyDefinition } from '~/types'
+import { eventDefinitionsTableLogicType } from './eventDefinitionsTableLogicType'
+import api from 'lib/api'
+
+import { isPostHogProp } from 'lib/components/PropertyKeyInfo'
+import { toParams } from 'lib/utils'
+interface DefinitionsPaginatedResponse {
+    count: number
+    next: string | null
+    previous: string | null
+}
+
+interface EventDefinitionsPaginatedResponse extends DefinitionsPaginatedResponse {
+    results: EventDefinition[]
+}
+
+export interface PropertyDefinitionWithExample extends PropertyDefinition {
+    example?: string
+}
+
+interface PropertyDefinitionsPaginatedResponse extends DefinitionsPaginatedResponse {
+    results: PropertyDefinitionWithExample[]
+}
+
+interface Filters {
+    event: string
+    properties: AnyPropertyFilter[]
+}
+
+export const EVENT_DEFINITIONS_PER_PAGE = 100
+export const PROPERTY_DEFINITIONS_PER_EVENT = 5
+
+export interface EventDefinitionsTableLogicProps {
+    key: string
+    syncWithUrl?: boolean
+}
+
+export const eventDefinitionsTableLogic = kea<
+    eventDefinitionsTableLogicType<
+        EventDefinitionsPaginatedResponse,
+        EventDefinitionsTableLogicProps,
+        Filters,
+        PropertyDefinitionsPaginatedResponse
+    >
+>({
+    path: (key) => ['scenes', 'data-management', 'events', 'eventDefinitionsTableLogic', key],
+    props: {} as EventDefinitionsTableLogicProps,
+    key: (props) => props.key || 'scene',
+    actions: {
+        loadEventDefinitions: (url: string | null = '') => ({ url }),
+        loadPropertiesForEvent: (definition: EventDefinition) => ({ definition }),
+        setFilters: (filters: Filters) => ({ filters }),
+    },
+    reducers: {
+        filters: [
+            {
+                event: '',
+                properties: [],
+            } as Filters,
+            {
+                setFilters: (_, { filters }) => filters,
+            },
+        ],
+    },
+    loaders: ({ values }) => ({
+        eventDefinitions: [
+            { count: 0, next: null, previous: null, results: [] } as EventDefinitionsPaginatedResponse,
+            {
+                loadEventDefinitions: async ({ url }, breakpoint) => {
+                    if (!url) {
+                        url = `api/projects/@current/event_definitions/?limit=${EVENT_DEFINITIONS_PER_PAGE}`
```

Comment:
It's better to use the builder methods in `api` for this. `@current` is something we actively want to avoid.